### DISCUSSION TITLE
GPG sign fails for unknown reason

### DISCUSSION THREAD

**Discussion by ixodie** [goreleaser/goreleaser#5044] [Q&A] [answered]
Hello,

I am using goreleaser to compile/sign a terraform provider.  Everything works up until the action tries to sign the artifacts:

[• creating                                       archive=dist/terraform-provider-name_0.0.3_darwin_amd64.zip
    • took: 13s
  • calculating checksums
  • signing artifacts
    • signing                                        cmd=gpg artifact=terraform-provider-name_0.0.3_SHA256SUMS signature=dist/terraform-provider-name_0.0.3_SHA256SUMS.sig
  ⨯ release failed after 7m24s               error=sign: gpg failed: exit status 2: gpg: skipped "50C580476CD7D5677196E0F7AEEE69224FXXXXXX": No secret key
gpg: signing failed: No secret key
Error: The process '/opt/hostedtoolcache/goreleaser-action/2.1.0/x[64](https://github.com/name/terraform-provider-name/actions/runs/10201190459/job/28222457416#step:5:65)/goreleaser' failed with exit code 1](url)

I have the GPG_PRIVATE_KEY and PASSPHRASE configured as repository secrets in my repo.

I have double checked the key value, it is exactly what I am using to sign my commits so I know it should work.

Any ideas?

**Comment by andrzej-zamojski** [comment]
Frankly speaking I have exactly same issue - tried already with different keys, with password and without and result is the same.

**Reply by ixodie** [reply]
Yeah I feel like I have tried everything, different keys, envvars that seem to be missing.  I'm really at a loss for how to troubleshoot this one.

**Comment by caarlos0** [comment, marked as answer]
that error seems to indicate the private key was not present...
